### PR TITLE
refactor: fix scalar udf part-2

### PR DESCRIPTION
**Pull request by kevinbheda**

### CODE REVIEW

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/FormatTimeInZone.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
      * @author oshank
      * @team GoFood
      */
+    //no data type change
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/FormatTimeInZone.java:26]: remove the comments.

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ArrayOperate.java [hunk 2]
```diff
@@ -38,16 +42,17 @@ public void open(FunctionContext context) throws Exception {
...
      * Given an Object Array, this UDF performs basic functions on the Array.
      *
-     * @param arrayElements the array elements
      * @param operationType the operation type
      * @param inputDataType the input data type
+     * @param arrayElements the array elements
      * @return the result of the aggregate
      */
-    public Object[] eval(Object[] arrayElements, String operationType, String inputDataType) {
+    public @DataTypeHint(value = "RAW", bridgedTo = List.class)
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ArrayOperate.java:50]: Could not find any Dagger using this and using something like `unnest` or `cardinality`. So changing the contract to Object[] to List<Object> should be fine here. However, I am trying a few things to support Object[] in the UDF contract. If that works we should change the contract to the old ones.

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/LinearTrend.java [hunk 3]
```diff
@@ -17,14 +19,16 @@
     /**
      * returns the gradient of the best fit line of the list of non-null demand values given the defined time window.
      *
-     * @param timestampsArray       the timestamps array
+     * @param localDateTimeArray    the timestamps array
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/LinearTrend.java:22]: Why did you convert timestampsArray to LocalDateTimeArray ? LocalDateTime would have timezone associated. Who gives it.

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ListContains.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
      * @author prakhar.m
      * @team DE
      */
+    //no data type change needed
```
Comment (arujit) [inline dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/ListContains.java:20]: remove this comment.

File: dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/ArrayOperateTest.java [hunk 5]
```diff
@@ -44,11 +46,8 @@ public void shouldComputeBasicOPerationsForArray() throws Exception {
         objects[4] = "a";
         ArrayOperate arrayOperate = new ArrayOperate();
         arrayOperate.open(functionContext);
-        Object[] result = arrayOperate.eval(objects, "distinct", "other");
-        Assert.assertEquals(3, result.length);
-        Assert.assertEquals("a", Array.get(result, 0));
-        Assert.assertEquals("b", Array.get(result, 1));
-        Assert.assertEquals("v", Array.get(result, 2));
+        List<Object> objectList = arrayOperate.eval("distinct", "other", objects);
```
Comment (arujit) [inline dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/ArrayOperateTest.java:49]: This looks a lot cleaner now.

File: dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/LinearTrendTest.java [hunk 6]
```diff
@@ -35,64 +38,63 @@ public void setup() {
 
     @Test
     public void shouldReturnGradientWhenNoMetricValueIsMissingInHopWindow() {
-        ArrayList<Timestamp> timestampsList = new ArrayList<>();
+        ArrayList<LocalDateTime> localDateTimes = new ArrayList<>();
```
Comment (arujit) [inline dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/LinearTrendTest.java:41]: Again why are we converting everything to LocalDateTime?

File: dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/LinearTrend.java (hunk 3 above)
Comment (kevinbheda) [reply dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/LinearTrend.java:22]: converted type for hopstarttime to localdatetime

**Pull request merged**